src/handlers/shared/start.ts

```diff
@@ -57,7 +59,7 @@ export async function start(context: Context, issue: Context["payload"]["issue"]
   const toAssign = [];
   // check max assigned issues
   for (const user of teammates) {
-    if (await handleTaskLimitChecks(user, context, maxConcurrentTasks, logger, sender.login)) {
+    if (await handleTaskLimitChecks(user, context, maxTask.limit, logger, sender.login)) {
```

Comment:
You have passed in the task limit for the `sender` for each call to this function essentially applying the same limit to all assignees in a team scenario as opposed to moving the call to `getUserRoleAndTaskLimit` into the checks function itself